service/src/main.rs

```diff
@@ -625,46 +625,42 @@ fn subscribe_to_parentchain_new_headers<E: EnclaveBase + Sidechain>(
 	}
 }
 
-/// Gets the amount of blocks to sync from the parentchain and feeds them to the enclave.
-///
+/// Fetches the amount of blocks to sync from the parentchain and feeds them to the enclave.
+/// Returns the latest synced block Header.
 ///
 pub fn sync_parentchain<E: EnclaveBase + Sidechain>(
 	enclave_api: &E,
 	api: &Api<sr25519::Pair, WsRpcClient>,
 	last_synced_header: Header,
 ) -> Header {
 	let tee_accountid = enclave_account(enclave_api);
-
 	trace!("Getting current head");
 	let curr_head: SignedBlock = api.last_finalized_block().unwrap().unwrap();
 	let head_block_number = curr_head.block.header.number;
 
-	let blocks_to_sync = get_blocks_to_sync(api, &last_synced_header, &curr_head);
-
-	println!("[+] Found {} block(s) to sync", blocks_to_sync.len());
-
 	let mut synced_header_until = last_synced_header;
+	loop {
+		let block_chunk_to_sync = get_block_chunk_to_sync(api, &synced_header_until, &curr_head);
+		println!("[+] Found {} block(s) to sync", block_chunk_to_sync.len());
+		if block_chunk_to_sync.is_empty() {
+			return synced_header_until
+		}
 
-	// only feed BLOCK_SYNC_BATCH_SIZE blocks at a time into the enclave to save enclave state regularly
-	for chunk in blocks_to_sync.chunks(BLOCK_SYNC_BATCH_SIZE as usize) {
 		let tee_nonce = api.get_nonce_of(&tee_accountid).unwrap();
-
-		if let Err(e) = enclave_api.sync_parentchain(chunk, tee_nonce) {
+		if let Err(e) = enclave_api.sync_parentchain(&block_chunk_to_sync[..], tee_nonce) {
```

Comment:
Is it a vector you're using here? Then you should probably use `as_slice()`, instead of the `&vec[..]` syntax.

Reply:
ok